lib/swaggerUtil.js

```diff
@@ -0,0 +1,139 @@
+'use strict'
+
+function localRefResolve (jsonSchema, externalSchemas) {
+  if (jsonSchema.type && jsonSchema.properties) {
+    // for the shorthand querystring/params/headers declaration
+    const propertiesMap = Object.keys(jsonSchema.properties).reduce((acc, headers) => {
+      const required = (jsonSchema.required && jsonSchema.required.indexOf(headers) >= 0) || false
+      const newProps = Object.assign({}, jsonSchema.properties[headers], { required })
+      return Object.assign({}, acc, { [headers]: newProps })
+    }, {})
+
+    return propertiesMap
+  }
+
+  // for oneOf, anyOf, allOf support in querystring/params/headers
+  if (jsonSchema.oneOf || jsonSchema.anyOf || jsonSchema.allOf) {
+    const schemas = jsonSchema.oneOf || jsonSchema.anyOf || jsonSchema.allOf
+    return schemas.reduce(function (acc, schema) {
+      const json = localRefResolve(schema, externalSchemas)
+      return { ...acc, ...json }
+    }, {})
+  }
+
+  // $ref is in the format: #/definitions/<resolved definition>/<optional fragment>
+  const localReference = jsonSchema.$ref.split('/')[2]
+  return localRefResolve(externalSchemas[localReference], externalSchemas)
+}
+
+// For supported keys read:
+// https://swagger.io/docs/specification/2-0/describing-parameters/
+function plainJsonObjectToSwagger2 (container, jsonSchema, externalSchemas) {
```

Comment:
If cookie support will be added, it will require changes in here as well. But note that swagger2 doesn't support them.

Reply:
I have split the two method.